src/app/recipes/components/recipe-details/recipe-details.component.ts

```diff
@@ -54,9 +64,113 @@ export class RecipeDetailsComponent implements OnInit, OnDestroy {
         } else {
             this.user.favoriteRecipes = [...this.user.favoriteRecipes, this.recipe.id];
         }
+
+        this.recipeBookmarked = this.isRecipeBookmarked();
+    }
+
+    isRecipeBookmarked() {
+       return this.user.favoriteRecipes.includes(this.recipe.id);
+    }
+
+    scroll(el: HTMLElement) {
+        el.scrollIntoView({behavior: 'smooth'});
+    }
+
+    onCheckIngredient(category: string, ingredient: string) {
```

Comment:
For the beginning - let's extract all this localStorage related logic to separate service and define a proper API for it